dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/file/FileOrchestrator.kt

```diff
@@ -35,7 +35,10 @@ internal class FileOrchestrator(
     private val recentReadDelayMs =
         filePersistenceConfig.recentDelayMs + (filePersistenceConfig.recentDelayMs / 20)
 
-    // region FileOrchestrator
+    override fun reset() {
+        previousFile = null
+        previousFileLogCount = 0
+    }
```

Comment:
How do we synchronize it with `getWritableFile()` and `getReadableFile()`?

Reply:
Actually it is a mistake here. I don't think I should reset the FileOrchestrator for the authorised folder.

Reply:
For the other orchestrator does not need to be synchronised as it is going to be used from one single thread (only for writing). No Uploader will use the IntermediaryFileOrchestrator

Reply:
We could actually instantiate a new one each time we provide a PendingDataProcessor but this will mean we're not going to be consistent. For the AuthorizedDataProcessor we need to share the FileOrchestrator with the DataUploader.

Reply:
In both ways it's slightly inconsistent: either we call `reset()` only in intermediate orchestrator OR we re-create it. This could be improved by introducing more interfaces for `FileOrchestrator`, but this would be IMO over-engineering.